src/librawspeed/decoders/ArwDecoder.cpp

```diff
@@ -145,6 +147,13 @@ RawImage ArwDecoder::decodeRawInternal() {
     return mRaw;
   }
 
+  if (7 == compression) {
+    DecodeLJpeg(raw);
+    // cropping of lossless compressed L files already done in Ljpeg decoder
```

Comment:
Not only L, but also APS-C/S35

Reply:
Question is, where in the whole pipeline does the encoding fail, it is not even executing the ARW decoder in case  of M/S images.